From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Thu, 19 Sep 2024 10:30:06 +0200
Subject: demux: mkv: handle mkv-use-chapter-codec option

For safety reasons some users may not want to handle any chapter codec.
The option was added 8697e6ef99ef0cec90bcf515052e5de1c09601ce but
never implemented.

(cherry picked from commit 4b479fb554cf1474fc549334459512b0b096dbcc)
---
 modules/demux/mkv/matroska_segment_parse.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/demux/mkv/matroska_segment_parse.cpp b/modules/demux/mkv/matroska_segment_parse.cpp
index 706733e..b9a2d27 100644
--- a/modules/demux/mkv/matroska_segment_parse.cpp
+++ b/modules/demux/mkv/matroska_segment_parse.cpp
@@ -1257,6 +1257,9 @@ void matroska_segment_c::ParseChapterAtom( int i_level, KaxChapterAtom *ca, chap
         {
             debug( vars, "ChapterProcess" );
 
+            if ( !var_InheritBool( vars.p_demuxer, "mkv-use-chapter-codec") )
+                return;
+
             chapter_codec_cmds_c *p_ccodec = NULL;
 
             for( size_t j = 0; j < cp.ListSize(); j++ )
